src/wasm-interpreter.h

```diff
@@ -2528,7 +2532,37 @@ template<typename GlobalManager, typename SubType> class ModuleInstanceBase {
 
   std::unordered_set<size_t> droppedSegments;
 
+  struct TableInterfaceInfo {
+    // The external interface in which the table is defined.
+    ExternalInterface* interface;
+    // The name the table has in that interface.
+    Name name;
+  };
+
+  TableInterfaceInfo getTableInterfaceInfo(Name name) {
+    auto* table = wasm.getTable(name);
+    if (table->imported()) {
+      auto& importedInstance = linkedInstances.at(table->module);
+      auto* tableExport = importedInstance->wasm.getExport(table->base);
```

Comment:
Do we need to handle link errors here or can we assume that the various modules link correctly?

Reply:
This is after validation and linking, so we can assume those things succeeded.

(This code is just moved around into a utility function - no known issues with it AFAIK.)